2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,476 @@
+nupkg) instead of expanding it into a folder.
+
+### Updating resources
+
+`Update-PSResource` will update all resources to most recent minor version by default.
+
+A `-UpdateTo` parameter has values `MinorVersion` (as default), `MajorVersion`, `PatchVersion`.
+`MajorVersion` allows updating to latest major version.
+`PatchVersion` allows updating to latest patch version (e.g. 6.2.1 to 6.2.4).
+
+If the installed resource is a pre-release, it will automatically update to
+latest prerelease or stable version (if available).
+
+### Publishing resources
+
+`Publish-PSResource` will supporting publishing modules, scripts and nupkgs.
+It will follow the same model as `Publish-Module` in v2.
+
+A `-DestinationPath` can be used to publish the resulting nupkg locally.
+`Publish-PSResource` can accept a path to the nupkg to publish to a repository.
+
+A `-Nuspec` parameter can be used to specify a nuspec file rather than relying
+on this module to produce one.
+
+A `-SkipDependenciesCheck` switch can be used to bypass the default check that
+all dependencies are present.
+
+### Listing installed resources
+
+`Get-PSResource` will list all installed resources with new `Type` column.
+
+```output
+Version  Name       Type    Repository  Description
+-------  ----       ----    ----------  -----------
+5.0.0    VSTeam     Module  PSGallery   Adds functionality for working with Visual …
+0.14.94  PSGitHub   Module  PSGallery   This PowerShell module enables integration …
+0.7.3    posh-git   Module  PSGallery   Provides prompt with Git status summary …
+1.0.1    PSAutoMute Script  PSGallery   Powershell script to Auto Mute you sound devices …
+```
+
+### Uninstalling resources
+
+`Uninstall-PSResource` will be available to remove installed resources.
+
+>[!NOTE]
+>Uninstalling dependencies automatically will be something to consider in the future.
+
+If a package is unable to be uninstalled because it is currently loaded, the cmdlet will emit a warning
+which enumerates the packages that were unable to be installed.
+We will explore a dependency clean parameter in a future version of PowerShellGet.
+
+When uninstalling the dependency check will be performed not only on the RequiredModules.Name,
+but also on RequiredModules.Version, which should be less-than-or-equal to the version of the module being
+uninstalled in order to be considered a dependency.
+
+### Proxy support
+
+Each cmdlet will have `-Proxy` and `-ProxyCredential` parameters.
+
+### PowerShellGallery status
+
+Upon failure to connect to PSGallery, the cmdlets should retrieve status from
+a well known REST API and return a more descriptive error message to the user.
+
+### Open Sourcing the module
+
+In order to open source PowerShellGet 3.0 we will fork the current PowerShellGet repo, rename it as PowerShellGetv2,
+and use the existing PowerShellGet repo as the primary support and development contact for PowerShellGet 3+.

```

Comment:
I so wish this had not been conducted this way as it is a terrible user experience for current users of PowerShellGetv2 including those like me that had it cloned already

Forking and calling it PowerShellGetv3 would have been a much better option, but generally a v3 branch & the branches off of this within the original repository generally makes the most sense for development & should be adhered to in future splits of modules where possible.